pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -512,7 +514,58 @@ func (configgen *ConfigGeneratorImpl) buildSidecarOutboundListeners(env *model.E
 	var tcpListeners, httpListeners []*xdsapi.Listener
 	// For conflict resolution
 	listenerMap := make(map[string]*listenerEntry)
-	for _, service := range services {
+
+	// The sidecarConfig if provided could filter the list of
+	// services/virtual services that we need to process. It could also
+	// define one or more listeners with specific ports. Once we generate
+	// listeners for these user specified ports, we will auto generate
+	// configs for other ports if and only if the sidecarConfig has an
+	// egressListener on wildcard port.
+	//
+	// Validation will ensure that we have utmost one wildcard egress listener
+	// occurring in the end
+
+	var catchAllListener *model.IstioListenerWrapper
+	var importedServices []*model.Service
+	var importedConfigs []model.Config
+	// Add listeners based on the config in the sidecar.EgressListeners if
+	// no Sidecar CRD is provided for this config namespace,
+	// push.SidecarScope will generate a default catch all egress listener.
+	for _, egressListener := range sidecarScope.EgressListeners {
+		if egressListener.IstioListener != nil &&
+			egressListener.IstioListener.Port != nil {
+			// We have a non catch all listener on some user specified port
+			// The user specified port may or may not match a service port.
+			// If it does not match any service port, then we expect the
+			// user to provide a virtualService that will route to a proper
+			// Service. This is the reason why we can't reuse the big
+			// forloop logic below as it iterates over all services and
+			// their service ports.
+
+			// TODO: complete implementation
+			continue
+		}
+
+		// This is a catch all egress listener. This should be the last
+		// egress listener in the sidecar Scope.
+		catchAllListener = egressListener
```

Comment:
If multi egress listeners specified, the first one takes effect.

Reply:
IF multiple egress listeners with no Port spec are specified, then the first one takes effect. As per our API, we say that specifying multiple egress listeners without any port is not allowed. Need to add validation code